typing/ctype.ml

```diff
@@ -2521,6 +2532,31 @@ let constrain_type_jkind ~fixed env ty jkind =
                   (Not_a_subjkind (ty's_jkind, jkind, sub_failure_reasons)))
              end
           in
+          let or_null ~fuel ty is_open modality =
+            let error =
+              Error (Jkind.Violation.of_ ~jkind_of_type
+                (Not_a_subjkind (ty's_jkind, jkind, sub_failure_reasons)))
+            in
+            let jkind = Jkind.apply_modality_r modality jkind in
+            match
+              Jkind.apply_or_null jkind
+            with
+            | Ok jkind ->
+              (match
+                loop ~fuel:(fuel - 1) ~expanded:false ty ~is_open
```

Comment:
This fuel deduction is redundant with the one at the call site of `or_null`. (Actually, we don't really need to deduct fuel at all in this case; the kinds stop us from doing it forever. But using fuel is ok.)